sktime/networks/scinet.py

```diff
@@ -0,0 +1,802 @@
+
+        if self.single_step_output_One:  # only output the N_th timestep.
+            if self.stacks == 2:
+                if self.concat_len:
+                    self.projection2 = nn.Conv1d(
+                        self.concat_len + self.pred_len, 1, kernel_size=1, bias=False
+                    )
+                else:
+                    self.projection2 = nn.Conv1d(
+                        self.seq_len + self.pred_len, 1, kernel_size=1, bias=False
+                    )
+        else:  # output the N timesteps.
+            if self.stacks == 2:
+                if self.concat_len:
+                    self.projection2 = nn.Conv1d(
+                        self.concat_len + self.pred_len,
+                        self.pred_len,
+                        kernel_size=1,
+                        bias=False,
+                    )
+                else:
+                    self.projection2 = nn.Conv1d(
+                        self.seq_len + self.pred_len,
+                        self.pred_len,
+                        kernel_size=1,
+                        bias=False,
+                    )
+
+        # For positional encoding
+        self.pe_hidden_size = input_dim
+        if self.pe_hidden_size % 2 == 1:
+            self.pe_hidden_size += 1
+
+        num_timescales = self.pe_hidden_size // 2
+        max_timescale = 10000.0
+        min_timescale = 1.0
+
+        log_timescale_increment = math.log(
+            float(max_timescale) / float(min_timescale)
+        ) / max(num_timescales - 1, 1)
+        # temp = torch.arange(num_timescales, dtype=torch.float32)
+        inv_timescales = min_timescale * torch.exp(
+            torch.arange(num_timescales, dtype=torch.float32) * -log_timescale_increment
+        )
+        self.register_buffer("inv_timescales", inv_timescales)
+
+        ### RIN Parameters ###
+        if self.RIN:
+            self.affine_weight = nn.Parameter(torch.ones(1, 1, input_dim))
+            self.affine_bias = nn.Parameter(torch.zeros(1, 1, input_dim))
+
+    def get_position_encoding(self, x):
+        """Generate position encoding for the input tensor `x`."""
+        max_length = x.size()[1]
+        position = torch.arange(
+            max_length, dtype=torch.float32, device=x.device
+        )  # tensor([0., 1., 2., 3., 4.], device='cuda:0')
+        # temp1 = position.unsqueeze(1)  # 5 1
+        # temp2 = self.inv_timescales.unsqueeze(0)  # 1 256
+        scaled_time = position.unsqueeze(1) * self.inv_timescales.unsqueeze(0)  # 5 256
+        signal = torch.cat(
+            [torch.sin(scaled_time), torch.cos(scaled_time)], dim=1
+        )  # [T, C]
+        signal = F.pad(signal, (0, 0, 0, self.pe_hidden_size % 2))
+        signal = signal.view(1, max_length, self.pe_hidden_size)
+
+        return signal
+
+    def forward(self, x):
+        """Perform the forward pass through the network."""
+        assert self.seq_len % (np.power(2, self.num_levels)) == 0
```

Comment:
also mention this in the docstring for `seq_len`

Reply:
@Sohaib-Ahmed21 I see that you mentioned this in the docstring for the network class `SCINet.forward`. This docstring is not visible to the user directly, we want this info to be in the docstring of `SCINetForecaster` so it is visible in the docs as you may see here: https://sktime--7400.org.readthedocs.build/en/7400/api_reference/auto_generated/sktime.forecasting.scinet.SCINetForecaster.html